core/src/types.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017 The Grin Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// TODO - what should this be called
+/// TODO - where should this live?
+#[derive(Debug)]
+pub enum CoreError {
+    /// TODO - rename this
+    GenericCoreError,
```

Comment:
I added this as a placeholder. We need a set of errors that can be shared between `pool` and `chain` and I wanted to avoid requiring one from the other.
1) This needs renaming and/or making more specific
2) Not sure where to put this (so put it in `types.rs` for now)

Reply:
I think we can have the adapter do the translation. So the chain throws its error. The pool trait requires its error as well. And the adapter implements the trait and translate the chain error into the pool one.